src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -93,16 +95,20 @@ public void Awake()
             DontDestroyOnLoad(gameObject);
             _this = this;
         }
-        
-        public void Start()
+        public void OnEnable()
         {
-            _rgServiceManager = RGServiceManager.GetInstance();
-            _rgBotManager = RGBotManager.GetInstance();
+            _this._mouseObserver = GetComponent<MouseInputActionObserver>();
         }
 
         private void OnDestroy()
         {
             _frameQueue?.CompleteAdding();
+            if (_fileWriteTasks.Count > 0)
+            {
+                RGDebug.LogInfo($"Waiting for unfinished file write tasks before stopping ["+ string.Join(",", _fileWriteTasks.Select(a=>a.Item1).ToArray())+ "]");
+            }
+            Task.WaitAll(_fileWriteTasks.Select(a=>a.Item2).ToArray());
+            _fileWriteTasks.Clear();
```

Comment:
With the new file writing approach, any worry about memory? I suppose we aren't in the realm of worrying about that yet?

Reply:
This approach isn't new.  I simply added the tracking to ensure all files are written before we move on to completing the recording as I found cases where that wasn't happening properly with big json/images